### PR TITLE
CMR-6778 Handle single datetime queries

### PR DESCRIPTION
**Pull request by jaybarra**
Update the temporal datetime handling to send a single valid datetime query to CMR.

### CODE REVIEW

**Comment by matthewhanson** [comment]
When providing a range, STAC API expects two dates delimited by '/'.  This currently works in prod:
https://cmr.earthdata.nasa.gov/stac/USGS_EROS/search?datetime=2020-01-01T00:00:00Z/2020-01-01T23:59:59Z


as does using a comma (although that is not STAC compliant)
https://cmr.earthdata.nasa.gov/stac/USGS_EROS/search?datetime=2020-01-01T00:00:00Z,2020-01-01T23:59:59Z

It looks as though changes here expect a comma only with the parseDatetime function so that this returns an error due to invalid datetime parameter:
http://localhost:3000/stac/USGS_EROS/search?datetime=2020-01-01T00:00:00Z/2020-01-01T23:59:59Z

**Comment by jaybarra** [comment]
Updated to handle slash delimited date ranges.

File: search/lib/convert/datetime.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+}
+
+/**
+ * Convert a datetime to a temporal range value.
+ * If a single datetime is given, duplicate it to be the end of the range.
+ * @param dateTimes
+ * @returns {*}
+ */
+function parseDateTimeHelper (dateTimes) {
+  let rc;
```
Comment (mschmele) [inline search/lib/convert/datetime.js:32]: Nitpick: this binding could have a much more descriptive name

File: search/lib/convert/datetime.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  }
+  return `${begin},${end}`;
+}
+
+/**
+ * Converts a datetime string to a CMR temporal range.
+ * If the value is a time, identity.
+ * If the value is a datetime => CMR range
+ */
+function parseDateTime (dt) {
```
Comment (mschmele) [inline search/lib/convert/datetime.js:75]: Nitpick: `dt` -> `dateTime`

File: search/lib/convert/datetime.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+  const dates = splitOnDelimiters(dt, [',', '/']);
+
+  let rc;
```
Comment (mschmele) [inline search/lib/convert/datetime.js:78]: Nitpick: same as above. Pick more descriptive name

File: search/tests/api/stac.spec.js [hunk 4]
```diff
@@ -76,3 +80,157 @@ describe('STAC Search', () => {
...
+
+  afterEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  describe('datetime', () => {
+    it('should query with range when given a single datetime', async () => {
+      request = createRequest({
+        params: { providerId: 'LPDAAC' },
+        query: { datetime: '2020-11-02T00:00:00Z' }
```
Comment (mschmele) [inline search/tests/api/stac.spec.js:106]: If there aren't already tests for handling malformed datetimes, it's worth adding them, as well as ones for something like:
`query: { datetime: '2020-11-02T00:00:00Z,' }` or `query: { datetime: '/2020-11-02T00:00:00Z' }`

**Review by mschmele — APPROVED** [review]

File: search/lib/convert/collections.js [hunk 5]
```diff
@@ -16,24 +17,20 @@ function cmrCollSpatialToExtents (cmrColl) {
       .map((rings) => rings[0])
       .map(pointStringToPoints)
       .reduce(addPointsToBbox, bbox);
-  }
-  if (cmrColl.points) {
+  } else if (cmrColl.points) {
```
[the remark this answers is not in the document]
Comment (jaybarra) [reply search/lib/convert/collections.js:20]: The way it was handled before could potentially overwrite the bbox if other geometries are passed. This way there is a single path the data can follow

**Pull request merged**